core/src/main/codegen/templates/Parser.jj

```diff
@@ -7666,6 +7666,9 @@ SqlPostfixOperator PostfixRowOperator() :
 |   < MEASURES: "MEASURES" >
 |   < MEMBER: "MEMBER" >
 |   < MERGE: "MERGE" > { beforeTableName(); }
+// TODO: I don't fully understand what beforeTableName does. It seems to modify the underlying lexical state somehow?
+// I think the safer option is to ommit it here, but I'm not sure.
+|   < MERGE_INTO: "MERGE_INTO" > //{ beforeTableName(); }
```

Comment:
Here is the source code and relevant sections:

```
void beforeTableName() {
      if (curLexState == BQID) {
        pushState();
        SwitchTo(BQHID);
      }
    }
```

```
BQID:    Identifiers are enclosed in back-ticks, escaped using backslash,
         e.g. `My \`Quoted\` Identifier`,
         and with the potential to shift into BQHID in contexts where table
         names are expected, and thus allow hyphen-separated identifiers as
         part of table names
BQHID:   Identifiers are enclosed in back-ticks, escaped using backslash,
         e.g. `My \`Quoted\` Identifier`
```

I think all this means is that because a table name is expected the types of identifiers that are accepted changed to be more lenient and allow for things specific to table names.

Reply:
I would include the `beforeTableName` because I do not believe there is any harm.